client/src/globals.ts

```diff
@@ -27,6 +27,7 @@ export const QUERY_PARAM_EXPLAIN_NEW_TAB = "explainNewTab";
  * and ending with "api". This setup ensures it starts capturing at the single-character dataroot and includes any optional subpaths.
  */
 const REGEX_PATHNAME = /(?<=\/)[^/]{1}\/(?:[^/]+\/)*[^/]+\/(?=api)/;
+const REGEX_PATHNAME_FOR_DATASET_SELECTOR = /(?<=\/)\w+\/[^/]+\/(?=api)/;
```

Comment:
maybe add a comment to explain the context why we need a specific REGEX for dataset selector instead of reusing the REFEX above?

Reply:
Great explanation! Thank you!